src/functional/ffront/type_info.py

```diff
@@ -158,13 +188,13 @@ def is_callable_for_args(
                 raise GTTypeError(f"Expected a function type, but got `{self.type}`.")
             return False
 
-        errors = function_signature_incompatibilities(self.type, args, kwargs)
+        errors = function_signature_incompatibilities(cast(FunctionType, self.type), args, kwargs)
         if raise_exception:
-            errors = list(errors)
-            if len(errors) > 0:
+            error_list = list(errors)
```

Comment:
Mypy deduces the type of `errors` to be `Iterator[str]`. From then on it complains about:
* assigning something else to it
* accessing members that `Iterator[str]` does not have such as Len

annotating `errors: list[str] | Iterator[str]` is no better because then it complains about both `len` and `next` because neither of them exists on all elements of the type union. More precisely, it would work but you have to put a type guard before every usage (which I think does not help readability).

I have not tried assigning to a temporary, deleting errors and then reassigning it (That would be taking it a bit too far in my opinion).

Alternatively we can also use `cast(errors)` or `# type: ignore` for all lines below the reassignment where mypy would complain.

I admit to taking the path of least resistance here.

Reply:
Thanks a lot for the explanation, I didn't know about the no-redefinition restriction. The solution makes sense.